datadog/dogstatsd/base.py

```diff
@@ -140,8 +173,21 @@ def __init__(self, host=DEFAULT_HOST, port=DEFAULT_PORT, max_buffer_size=50, nam
             self.port = int(port)
             transport = "udp"
 
+        self.telemetry_destination = True
```

Comment:
I'm not a fan of having 2 boolean to reflect the state of the telemetry (like `if self.telemetry and self.telemetry_destination`. I think we always compute the `telemetry_destination` and only check `self.telemetry` in the rest of the code.

By default `telemetry_destination` would be equal to host+port/socket used for metrics.

Reply:
I can try to just use `telemetry_socket`.